src/main/java/com/axonivy/connector/aws/authentication/CanonicalRequest.java

```diff
@@ -38,8 +41,14 @@ private void appendPath() {
     if (path == null || path.isEmpty()) {
       path = "/";
     }
-    builder.append(path);
-    builder.append('\n');
+
+    try {
+      var encodedPath = new URI(null, null, path, null).toASCIIString();
```

Comment:
Is there no better way to only encode the path?

Reply:
nope. the jdk has no better options. there are libraries out-there. everybody is cooking his own little soup.